torch/csrc/jit/codegen/cuda/kernel.cpp

```diff
@@ -44,7 +44,7 @@ struct ExtractSizeStride {
       TORCH_CHECK(b_dim >= o_dim);
       for (int i = 0; i < b_dim; i++) {
         sizes.push_back(broadcasted_size->at(i));
-        int index = i + o_dim - b_dim;
+        int index = i + o_dim - b_dim; // ??? 
```

Comment:
@jjsjann123 can provide insight here.

Reply:
I'd love to understand this better. To be fair I didn't spend a lot of time looking into it, but perhaps more explicit variable names and/or some comments would make it easier to read.

Reply:
I should have commented this.
PyTorch follows numpy broadcasting rule. https://numpy.org/doc/stable/user/basics.broadcasting.html
So in case where the rank of two operators differ, we align them on the higher dimensions -> hence the offset `o_dim-b_dim` to the index here.

Reply:
thanks!